### PR TITLE
add go-apidiff make target and CI check

### LINKED ISSUE #54: Add go-apidiff to CI
What do we think about running my joelanford/go-apidiff in CI to give us signal about PRs that introduce new APIs or break existing APIs?

### PR DESCRIPTION
**Pull request by joelanford**
Signed-off-by: Joe Lanford <joe.lanford@gmail.com>

Closes #54

### CODE REVIEW

File: .github/workflows/sanity.yaml
Comment (joelanford) [inline .github/workflows/sanity.yaml:32]: Rather than hardcoding to 1.43 here (and risking getting out of sync with what's in our `hack/tools/go.mod`), I thought "let's just read our go.mod and use that!"

Comment (timflannagan) [reply .github/workflows/sanity.yaml:32]: Do we need a `needs` keyword here if we're reliant on the output of parsing the go.mod in the previous job step?

**Review by timflannagan — APPROVED** [review]
lgtm - It would be nice if we could enable caching for the tool-related binaries so we don't need to always re-populate the module cache every time a CI action is run.

**Comment by joelanford** [comment]
/hold
Trying a few things out.

I agree caching is good. I added auto-caching to the go-apidiff action last week after being inspired by golangci-lint's action, so we could use that and do the same sort of version detection that I added with go-apidiff. (That's the current state of this PR)

Alternatively, we could also just continue using `make go-apidiff` and I could add an explicit caching step in the `go-apidiff` job.

**Comment by joelanford** [comment]
Closing - we don't need this until we're closer to a point where API stability matters.

**Pull request closed without merging**
